yocs_cmd_vel_mux/include/yocs_cmd_vel_mux/cmd_vel_subscribers.hpp

```diff
@@ -62,25 +59,27 @@ class CmdVelSubscribers
   {
   public:
     unsigned int           idx;          /**< Index; assigned according to the order on YAML file */
-    std::string            name;         /**< Descriptive name */
-    ros::Subscriber        subs;         /**< The subscriber itself */
+    std::string            name;         /**< Descriptive name; must be unique to this subscriber */
     std::string            topic;        /**< The name of the topic */
+    ros::Subscriber        subs;         /**< The subscriber itself */
     ros::Timer             timer;        /**< No incoming messages timeout */
     double                 timeout;      /**< Timer's timeout, in seconds  */
     unsigned int           priority;     /**< UNIQUE integer from 0 (lowest priority) to MAX_INT */
     std::string            short_desc;   /**< Short description (optional) */
     bool                   active;       /**< Whether this source is active */
 
-    CmdVelSubs(unsigned int idx) : idx(idx), active(false) {};
+    CmdVelSubs(unsigned int idx) : idx(idx), active(false) { };
+    ~CmdVelSubs() { }
 
+    /** Fill attributes with a YAML node content */
     void operator << (const YAML::Node& node);
   };
 
   CmdVelSubscribers() : allowed(VACANT) { }
   ~CmdVelSubscribers() { }
 
-  std::vector<CmdVelSubs>::size_type size() { return list.size(); };
-  CmdVelSubs& operator [] (unsigned int idx) { return list[idx]; };
+  std::vector<std::shared_ptr<CmdVelSubs>>::size_type size() { return list.size(); };
+  std::shared_ptr<CmdVelSubs>& operator [] (unsigned int idx) { return list[idx]; };
```

Comment:
The reason vectors do the reference return is so that you can put the function on the left hand side of an assignment, e.g.

```
   a[i] = 7;
```

There is an example [here](http://www.cs.technion.ac.il/users/yechiel/c++-faq/returning-refs.html). Also have to remember this functionality was required before shared pointers were around. I'd say if you're not needing that special functionality, the wierd things that can bamboozle your brain are better avoided by just using a shared pointer if c++11 is being used.

Question: do you actually use this function? I suspect you're just iterating over the thing to populate the mux. Just return iterator handles or subclass the vector?